src/components/structures/auth/Registration.tsx

```diff
@@ -332,8 +332,6 @@ export default class Registration extends React.Component<IProps, IState> {
                 }
             } else if (response.errcode === "M_USER_IN_USE") {
                 errorText = _t("Someone already has that username, please try another.");
-            } else if (response.errcode === "M_THREEPID_IN_USE") {
-                errorText = _t("That e-mail address is already in use.");
```

Comment:
It feels wrong to just not show any text, doesn't it?

Reply:
`errorText` is not empty. it contains error from server and we are just fill it by better error message.
Server error message is better than showing wrong error message.

https://github.com/matrix-org/matrix-react-sdk/blob/b68bd77505362a9764a9306c1355fef69339e18c/src/components/structures/auth/Registration.tsx#L302

Reply:
Right, but afaik the server error wouldn't be translated, so I think it would be better to still specify an error message

Reply:
something like this message?
`That e-mail address or phone number is already in use`

Reply:
Unless there is a way to figure out which one it actually was, then yes

Reply:
This is response from server
`{errcode: "M_THREEPID_IN_USE", error: "Phone number is already in use"}`

We can handle it by parsing error from response but parsing this can be troublesome later when the server changes the response.

for example if `error.toLowerCase().includes("phone")` then showing phone error and else showing email error.

Reply:
Yeah, I don't think parsing the server error is the answer, unless the client itself knows, we should show a more generic error

Reply:
Done,
just we must run translate bot for other languages.